### PR TITLE
Fixes race condition in sign-in

### PR DESCRIPTION
**Pull request by dmohs**
Better use of the Google JavaScript listener.

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs [hunk 1]
```diff
@@ -84,10 +85,26 @@
...
+      (aget "currentUser")
+      (js-invoke
+       "listen" (fn [u]
+                  (doseq [[_ on-change] @user-listeners]
+                    (on-change u))))))
 
 (defn get-access-token []
-  (-> @google-auth2-instance (.-currentUser) (.get) (.getAuthResponse) (.-access_token)))
+  (-> @google-auth2-instance
+      (aget "currentUser") (js-invoke "get") (js-invoke "getAuthResponse") (aget "access_token")))
```
Comment (zarsky-broad) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs:107]: Is this really better? It seems a lot more complicated.

Comment (dvoet) [reply src/cljs/org/broadinstitute/firecloud_ui/utils.cljs:107]: the old way may be been simpler but it did not work... working is better.

**Review by dvoet — APPROVED** [review]
👍

**Comment by davidangb** [comment]
Does this fix GAWB-1394?

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs (hunk 1 above)
Comment (dmohs) [reply src/cljs/org/broadinstitute/firecloud_ui/utils.cljs:107]: This pattern is necessary to support advanced compilation. I had to turn that off a while back, but I'd like to collect more data before we decide to abandon it.

**Comment by dmohs** [comment]
My suspicion is that this does fix https://broadinstitute.atlassian.net/browse/GAWB-1394 but I'll want to test that independently.

**Pull request merged**